lib/improver/tests/spotdata_new/spotdata_new/test_SpotLapseRateAdjust.py

```diff
@@ -0,0 +1,308 @@
+
+         Lapse rate  Orography  Temperatures (not used directly)
+          (x DALR)
+
+            A B C      A B C        A   B   C
+
+        a   2 1 1      1 1 1       270 270 270
+        b   1 2 1      1 4 1       270 280 270
+        c   1 1 2      1 1 1       270 270 270
+
+        Spot
+        (note the neighbours are identified with the A-C, a-c indices above)
+
+         Site  Temperature Altitude  Nearest    DZ   MinDZ      DZ
+                                     neighbour       neighbour
+
+          0        280        3      Ac         2    Bb         -1
+          1        270        4      Bb         0    Bb          0
+          2        280        0      Ca        -1    Ca         -1
+
+
+        """
+        # Set up lapse rate cube
+        attributes = {
+            'mosg__grid_domain': 'uk',
+            'mosg__grid_type': 'standard',
+            'mosg__grid_version': '1.2.0',
+            'mosg__model_configuration': 'uk_det'}
+
+        lapse_rate_data = np.ones(9).reshape(3, 3).astype(np.float32) * DALR
+        lapse_rate_data[0, 2] = 2 * DALR
+        lapse_rate_data[1, 1] = 2 * DALR
+        lapse_rate_data[2, 0] = 2 * DALR
+        self.lapse_rate_cube = set_up_variable_cube(lapse_rate_data,
+                                                    name="lapse_rate",
+                                                    units="K m-1",
+                                                    attributes=attributes,
+                                                    spatial_grid="equalarea")
+
+        # Set up neighbour and spot diagnostic cubes
+        y_coord, x_coord = construct_xy_coords(3, 3, "equalarea")
+        y_coord = y_coord.points
+        x_coord = x_coord.points
+
+        # neighbours, each group is for a point under two methods, e.g.
+        # [ 0.  0.  0.] is the nearest point to the first spot site, whilst
+        # [ 1.  1. -1.] is the nearest point with minimum height difference.
+        neighbours = np.array([[[0., 0., 2.],
+                                [1., 1., -1.]],
+                               [[1., 1., 0.],
+                                [1., 1., 0.]],
+                               [[2., 2., -1.],
+                                [2., 2., -1.]]])
+        altitudes = np.array([3, 4, 0])
+        latitudes = np.array([y_coord[0], y_coord[1], y_coord[2]])
+        longitudes = np.array([x_coord[0], x_coord[1], x_coord[2]])
+        wmo_ids = np.arange(3)
+        grid_attributes = ['x_index', 'y_index', 'vertical_displacement']
+        neighbour_methods = ['nearest', 'nearest_minimum_dz']
+        self.neighbour_cube = build_spotdata_cube(
+            neighbours, 'grid_neighbours', 1, altitudes, latitudes,
+            longitudes, wmo_ids, grid_attributes=grid_attributes,
+            neighbour_methods=neighbour_methods)
+        self.neighbour_cube.attributes = attributes
+
+        time = dt(2017, 11, 10, 4, 0)
+        time_bounds = None
+        frt = dt(2017, 11, 10, 0, 0)
```

Comment:
You could make these relative to the variable cube (for resilience), but that might not make for pretty code - so up to you.

Reply:
I don't understand I'm afraid.

Reply:
Done.